src/library_sockfs.js

```diff
@@ -280,6 +280,13 @@ mergeInto(LibraryManager.library, {
 
         function handleMessage(data) {
           assert(typeof data !== 'string' && data.byteLength !== undefined);  // must receive an ArrayBuffer
+
+          // An empty ArrayBuffer will emit a pseudo disconnect event
```

Comment:
If both client and server are using emscripten then there is no way to trigger this (As far as I know) but since emscripten's socket implementation is based on WebSocket, that means that it may be desirable for some users that they might use a server which is using node plus some websocket module. In this case it can be easily triggered:

```
let wsServer = require('ws').Server;
let server = new wsServer({ port: 8084 });

server.on('connection', function connection(socket) {
  console.log("Connection!");
  socket.on('message', function incoming(message) {
    console.log('received: %s', message);
  });
  socket.on('close', function closed() {
    console.log("disconnected");
  });
  socket.send(new Buffer("", 'utf-8')); // We are sending an empty ArrayBuffer/Buffer which will trigger the "bug"
});
```